api/handlers.go

```diff
@@ -743,24 +743,25 @@ func postImagesCreate(c *context, w http.ResponseWriter, r *http.Request) {
 		errorFound := false
 		nonOSErrorFound := false
 		successfulPull := false
-		callback := func(what, status string, err error) {
-			if err != nil {
+		callback := func(msg cluster.JSONMessageWrapper) {
+			msg.Msg.ID = msg.EngineName
+			if msg.Msg.Status != "" {
+				msg.Msg.Status = fmt.Sprintf("Pulling %s... : %s", image, msg.Msg.Status)
+			} else {
+				msg.Msg.Status = fmt.Sprintf("Pulling %s...", image)
+			}
+			if msg.Success {
+				successfulPull = true
+			}
```

Comment:
We want `successfulPull` to be true if `msg.Success` is true for any message, not all messages. If we did `successfulPull = msg.Success`, then `successfulPull` would just be set to the value of the most recent `msg.Success`. I've added a comment to clarify.